cupy/_environment.py

```diff
@@ -135,14 +135,43 @@ def _get_cub_path():
 
 
 def _setup_win32_dll_directory():
-    # Setup DLL directory to load CUDA Toolkit libs on Windows & Python 3.8+.
-    if sys.platform.startswith('win32') and (3, 8) <= sys.version_info:
+    # Setup DLL directory to load CUDA Toolkit libs and shared libraries
+    # added during the build process.
+    if sys.platform.startswith('win32'):
+        # Path to the CUDA Toolkit binaries
         cuda_path = get_cuda_path()
-        if cuda_path is None:
-            raise RuntimeError('CUDA path could not be detected.')
-        cuda_bin_path = os.path.join(cuda_path, 'bin')
-        _log('Adding DLL search path: {}'.format(cuda_bin_path))
-        os.add_dll_directory(cuda_bin_path)
+        if cuda_path is not None:
+            cuda_bin_path = os.path.join(cuda_path, 'bin')
+        else:
+            cuda_bin_path = None
+            warnings.warn(
```

Comment:
is it safe to change from error to warning?

Reply:
This mitigates the error anaconda users will face when Python 3.8+. https://github.com/cupy/cupy/issues/3403